src/group_theory/perm/basic.lean

```diff
@@ -287,6 +287,18 @@ equiv.ext $ λ ⟨x, hx⟩, by { dsimp [subtype_perm, of_subtype],
 
 @[simp] lemma default_perm {n : Type*} : default (equiv.perm n) = 1 := rfl
 
+/-- Permutations on a subtype are equivalent to permutations on the original type that fix pointwise
+the rest. -/
+@[simps] protected def subtype_equiv_subtype_perm (p : α → Prop) [decidable_pred p] :
+  perm (subtype p) ≃ {f : perm α // ∀ a, ¬p a → f a = a} :=
+{ to_fun := λ f, ⟨f.of_subtype, λ a, f.of_subtype_apply_of_not_mem⟩,
+  inv_fun := λ f, (f : perm α).subtype_perm
+    (λ a, ⟨decidable.not_imp_not.1 $ λ hfa, (f.val.injective (f.prop _ hfa) ▸ hfa),
+    decidable.not_imp_not.1 $ λ ha hfa, ha $ f.prop a ha ▸ hfa⟩),
+  left_inv := equiv.perm.subtype_perm_of_subtype,
+  right_inv := λ f,
+    subtype.ext (equiv.perm.of_subtype_subtype_perm _ $ λ a, not.decidable_imp_symm $ f.prop a) }
+
```

Comment:
I had these written for [this comment here](https://github.com/leanprover-community/mathlib/pull/7526#discussion_r693339646), but I think they belong better here
```suggestion
lemma perm.subtype_equiv_left_apply {α : Type*} (p : α → Prop) [decidable_pred p]
  (f : perm (subtype p)) (a : α) (h : p a) : perm.subtype_equiv p f a = f ⟨a, h⟩ :=
  perm.of_subtype_apply_of_mem f h

lemma perm.subtype_equiv_right_apply {α : Type*} (p : α → Prop) [decidable_pred p]
  (f : perm (subtype p)) (a : α) (h : ¬ p a) : perm.subtype_equiv p f a = a :=
  f.of_subtype_apply_of_not_mem h
```

Reply:
Do you think we really need those? Can't `f.of_subtype_apply_of_mem h` and ` `f.of_subtype_apply_of_not_mem h` already work for your purposes, with maybe a `rw equiv.perm.subtype_equiv_apply_coe`? or is it not rewriting as you would want to?

Reply:
It rewrites fine if I unfold first, but TBH I mostly just wrote them because, looking at data/equiv/basic.lean, a lot of the defs have corresponding lemmas like this. Seemed like following suit.

Reply:
Okay sure 👍

Reply:
Oh well, these aren't correct simp lemmas.